main.c

```diff
@@ -611,11 +611,12 @@ void *socks5_thread(void *thread_data) {
 		i = (sd >= 0);
 	} else {
 		snprintf(tport, MINIBUF_SIZE, "%d", ntohs(port));
+		char *thost_noport = strdup(thost);
 		strlcat(thost, ":", HOST_BUFSIZE);
 		strlcat(thost, tport, HOST_BUFSIZE);
 
 		tcreds = new_auth();
-		sd = proxy_connect(tcreds, "/", thost);
+		sd = proxy_connect(tcreds, thost, thost_noport);
```

Comment:
Here the line becomes:

```
sd = proxy_connect(tcreds, thost, hostname);
```

Having `hostname`, lines from 621 to 625 can be removed and line 626 becomes

```
sd = host_connect(hostname, ntohs(port));
```

Reply:
Yeah, I was already thinking the existing code was odd with adding and removing it again :) This is great, thanks!